pilot/pkg/config/coredatamodel/controller.go

```diff
@@ -238,18 +242,21 @@ func (c *Controller) Apply(change *mcpclient.Change) error {
 	return nil
 }
 
-// HasSynced is not implemented, waiting for the following issue
-// pending https://github.com/istio/istio/issues/7947
+// HasSynced returns true if the first batch of items has been popped
 func (c *Controller) HasSynced() bool {
-	// TODO:The configStore already populated with all the keys to avoid nil map issue
-	if len(c.configStore) == 0 {
-		return false
-	}
-	for _, descriptor := range c.ConfigDescriptor() {
-		if _, ok := c.configStore[descriptor.Type]; !ok {
-			return false
+	var notReady []string
+
+	c.syncedMu.Lock()
+	for messageName, synced := range c.synced {
+		if !synced {
+			notReady = append(notReady, messageName)
 		}
+	}
+	c.syncedMu.Unlock()
 
+	if len(notReady) > 0 {
+		log.Infof("Configuration not synced: first push for %v not received", notReady)
```

Comment:
Should this be Info or debug?

Reply:
This is hooked up directly to the app readiness probes which are invoked every ~30 seconds after a 5 second delay. It seems reasonable to keep this as INFO since it effects the overall health/readiness of the service and is infrequent.

Reply:
What happens if a config fails ( for example there is no CRD for that resource in the install - like in a 'a-la-carte' config ) ?

Reply:
I have another PR which adds the CRD check during startup. It relies on a static schema for registering types. Dynamically configured types/CRD is targeted for 1.2 (see https://github.com/istio/istio/issues/9063).